### PR TITLE
Use dropdowns for filtering mapselect

### PR DESCRIPTION
**Pull request by gunchleoc**
We are running out of horizontal UI space, so grouped most of the map tags into dropdowns 

map_filters

### CODE REVIEW

File: data/maps/Three_Warriors.wmf/elemental [hunk 1]
```diff
@@ -9,4 +9,4 @@ name=_"Three Warriors"
 author="Quappo"
 descr=_"Mighty mountains divide this land into three fertile valleys. But a path connects the valleys and thus a fight seems inescapable!"
 hint=_"Water can be found around the lakes."
-tags="ffa,official"
+tags="ffa,balanced,official"
```
Comment (stonerl) [inline data/maps/Three_Warriors.wmf/elemental]: `tags="balanced,ffa,official"`

File: data/maps/Wideworld.wmf/elemental [hunk 2]
```diff
@@ -9,4 +9,4 @@ name=_"Wide World"
 author="Einstein13"
 descr=_"The world is not enough! If you want to win, you have to send some ships. But remember: your opponents are very close to you."
 hint=
-tags="seafaring,official"
+tags="seafaring,official,balanced"
```
Comment (stonerl) [inline data/maps/Wideworld.wmf/elemental]: `tags="balanced,official,seafaring"`

File: src/ui_basic/dropdown.h [hunk 3]
```diff
@@ -141,6 +141,10 @@ class BaseDropdown : public NamedPanel {
 	void set_size(int nw, int nh) override;
 	void set_desired_size(int w, int h) override;
 
+	/// Expand display button to make enough room for each entry's text. Call this before adding yna
```
Comment (stonerl) [inline src/ui_basic/dropdown.h]: any

**Review by stonerl — DISMISSED** [review]
Some minor nits. But overall LGTM :-)

**Comment by gunchleoc** [comment]
Thanks for the review :)

**Pull request merged**